diff --git a/src/library/generator.stockham.cpp b/src/library/generator.stockham.cpp
index f100c11..9f78b2b 100644
--- a/src/library/generator.stockham.cpp
+++ b/src/library/generator.stockham.cpp
@@ -3885,7 +3885,7 @@ namespace StockhamGenerator
 							if(inInterleaved || inReal)
 							{
 								if(!rcSimple) {	str += "lwbIn2 = gbIn + iOffset2;\n\t"; }
-								str += "lwbIn = gbIn + iOffset;\n\t"; 
+								str += "lwbIn = (__global float2 *)gbIn + iOffset;\n\t"; 
 							}
 							else
 							{
@@ -3961,7 +3961,7 @@ namespace StockhamGenerator
 						{
 							if(inInterleaved)
 							{
-								str += "lwbIn = gbIn + iOffset;\n\t";
+								str += "lwbIn = (__global float2 *)gbIn + iOffset;\n\t";
 							}
 							else
 							{
